C/CLFFT/build_tarballs.jl

```diff
@@ -0,0 +1,55 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "CLFFT"
+version = v"2.12.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/clMathLibraries/clFFT.git", "1e4833f060976971c4df4b54b1b9ad1620aaf1fb"),
+    FileSource("https://patch-diff.githubusercontent.com/raw/clMathLibraries/clFFT/pull/245.patch",
+               "b78e80258bd5dd41572dfad8f4c8fe3b0192d7d4f3bdaa85aae2d59f61184a88")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+
+install_license ./clFFT/LICENSE
+
+patch ./clFFT/src/CMakeLists.txt 245.patch
+
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DBUILD_EXAMPLES=OFF -S ./clFFT/src -B ./clFFT/build
+cmake --build ./clFFT/build --target install -j${nproc}
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("aarch64", "macos"),
+    Platform("x86_64", "macos")
+]
+
+platforms = expand_cxxstring_abis(platforms)
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libclFFT", :libclfft)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency(PackageSpec(; name="OpenCL_Headers_jll", version=v"2022.09.23")),
+    Dependency("OpenCL_jll", compat="2022.09.23"),
```

Comment:
Regarding the OpenCL_jll that I compiled exactly with OpenCL_Headers_jll v2022.03.23, maybe it is a good idea to specify the version as I did with OpenCL_Headers_jll above?

Reply:
What do you mean exactly?

Reply:
I mean the recipe for building OpenCL_jll depends on OpenCL_Headers with a specific version. Does it matter?

Reply:
Aren't you already using the same version as in https://github.com/JuliaPackaging/Yggdrasil/blob/e4b3cca311849b09ff9cd959873883c0b4cc6554/O/OpenCL/build_tarballs.jl#L38?

Reply:
My question is: we built OpenCL_jll with a specific version of OpenCL_Headers_jll. Now in this new recipe for CLFFT_jll we are saying that we need a specific version of OpenCL_Headers_jll but only a compat version of OpenCL_jll to build. From what I understand if someone releases a patch for OpenCL_jll this will trigger a new build of CLFFT_jll where the versions of OpenCL_jll and OpenCL_Headers_jll are not synced? Am I over-complicating things?

Reply:
> Am I over-complicating things?

Yes. A bug fix only release is unlikely to change the API of a package, almost by definition. But I'm still missing what's your concern: next time opencl will be updated, the corresponding headers package can be updated as well, so I'm missing what's the problem.

Reply:
Got it, I am just asking to double check if my understanding is correct. Marking as solved 👍🏽